container.go

```diff
@@ -272,6 +280,101 @@ func (c *container) Update(ctx context.Context, opts ...UpdateContainerOpts) err
 	return nil
 }
 
+func (c *container) Checkpoint(ctx context.Context, ref string, opts ...CheckpointOpts) (Image, error) {
+	index := &ocispec.Index{
+		Versioned: ver.Versioned{
+			SchemaVersion: 2,
+		},
+		Annotations: make(map[string]string),
+	}
+	copts := &options.CheckpointOptions{
+		Exit:                false,
+		OpenTcp:             false,
+		ExternalUnixSockets: false,
+		Terminal:            false,
+		FileLocks:           true,
+		EmptyNamespaces:     nil,
+	}
+	info, err := c.Info(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	img, err := c.Image(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx, done, err := c.client.WithLease(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer done(ctx)
+
+	// add image name to manifest
+	ir := bytes.NewReader([]byte(img.Name()))
+	idesc, err := writeContent(ctx, c.client.ContentStore(), images.MediaTypeContainerd1CheckpointImageName, info.ID+"-image-name", ir)
+	if err != nil {
+		return nil, err
+	}
+	idesc.Platform = &ocispec.Platform{
+		OS:           runtime.GOOS,
+		Architecture: runtime.GOARCH,
+	}
+	index.Manifests = append(index.Manifests, idesc)
+
+	// add runtime info to index
+	rr := bytes.NewReader([]byte(info.Runtime.Name))
+	rdesc, err := writeContent(ctx, c.client.ContentStore(), images.MediaTypeContainerd1CheckpointRuntimeName, info.ID+"-runtime-name", rr)
```

Comment:
an we combine the runtime name and options as a checkpoint type.  I'm not sure about single string descriptors but it maybe more tricky since the options are a proto, we could maybe just marshal it to json instead, not sure

Reply:
Ok, i think we need to keep the runtime name as an annotation as well, then we would know how to unmarshal the options.  sorry if i told you differently before.  I think this can still be an opt.

Reply:
ok sounds good